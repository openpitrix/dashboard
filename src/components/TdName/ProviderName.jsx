import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { Icon } from 'components/Base';
import styles from './index.scss';

export default class ProviderName extends Component {
  static propTypes = {
    className: PropTypes.string,
    name: PropTypes.string,
    provider: PropTypes.string
  };

  static defaultProps = {
    provider: 'qingcloud'
  };

  getShowProvider = name => {
    const nameMap = {
      qingcloud: 'QingCloud',
      aws: 'AWS',
      kubernetes: 'Kubernetes'
    };
    if (nameMap[name]) {
      return nameMap[name];
    }
    return name;
  };

  render() {
    const { name, provider, className } = this.props;

    return (
      <div className={classnames(styles.repoName, className)} title={name}>
        <Icon name={provider} size={20} className={styles.icon} type="dark" />
        {this.getShowProvider(name)}
      </div>
    );
  }
}
