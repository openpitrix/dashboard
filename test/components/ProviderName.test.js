import ProviderName from 'components/TdName/ProviderName';

describe('Component ProviderName', () => {
  it('basic render', () => {
    const wrapper = render(<ProviderName />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has find provider', () => {
    const wrapper = render(<ProviderName name="qingcloud" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
