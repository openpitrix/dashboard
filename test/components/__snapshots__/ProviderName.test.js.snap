// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProviderName basic render 1`] = `
<div
  class="repoName"
>
  <span
    class="icon icon-qingcloud icon"
    style="width:20px;height:20px"
  >
    <svg
      class="qicon qicon-qingcloud qicon-dark"
    >
      <use
        href="#qui-qingcloud"
      />
    </svg>
  </span>
</div>
`;

exports[`Component ProviderName has find provider 1`] = `
<div
  class="repoName"
  title="qingcloud"
>
  <span
    class="icon icon-qingcloud icon"
    style="width:20px;height:20px"
  >
    <svg
      class="qicon qicon-qingcloud qicon-dark"
    >
      <use
        href="#qui-qingcloud"
      />
    </svg>
  </span>
  QingCloud
</div>
`;
